test failures

```python
=============================================== FAILURES ===============================================
___________________________________________ test_all_labels ____________________________________________
[gw6] linux -- Python 3.4.3 /usr/bin/python3.4
def test_all_labels():
        for label in LABELS:
>           assert decode(b'', label) == ''
E           assert ('', <Encoding iso-8859-4>) == ''
E            +  where ('', <Encoding iso-8859-4>) = decode(b'', 'iso88594')

webencodings/tests.py:50: AssertionError
_________________________________________ test_x_user_defined __________________________________________
[gw1] linux -- Python 3.4.3 /usr/bin/python3.4
def test_x_user_defined():
        encoded = b'2,\x0c\x0b\x1aO\xd9#\xcb\x0f\xc9\xbbt\xcf\xa8\xca'
        decoded = '2,\x0c\x0b\x1aO\uf7d9#\uf7cb\x0f\uf7c9\uf7bbt\uf7cf\uf7a8\uf7ca'
        encoded = b'aa'
        decoded = 'aa'
>       assert decode(encoded, 'x-user-defined') == decoded
E       assert ('aa', <Encoding x-user-defined>) == 'aa'
E        +  where ('aa', <Encoding x-user-defined>) = decode(b'aa', 'x-user-defined')

webencodings/tests.py:152: AssertionError
_____________________________________________ test_decode ______________________________________________
[gw5] linux -- Python 3.4.3 /usr/bin/python3.4
def test_decode():
>       assert decode(b'\x80', 'latin1') == '€'
E       assert ('€', <Encoding windows-1252>) == '€'
E        +  where ('€', <Encoding windows-1252>) = decode(b'\x80', 'latin1')

webencodings/tests.py:77: AssertionError
================================== 3 failed, 5 passed in 1.67 seconds ==================================
```

--- a/tox.ini
+++ b/tox.ini
@@ -2,7 +2,7 @@
 python_files=test*.py
 
 [tox]
-envlist = py26, py27, py32, py33, pypy
+envlist = py26, py27, py32, py33, py34, py35, pypy, pypy3
 
 [testenv]
 deps=pytest

--- a/webencodings/tests.py
+++ b/webencodings/tests.py
@@ -47,7 +47,7 @@ def test_labels():
 
 def test_all_labels():
     for label in LABELS:
-        assert decode(b'', label) == ''
+        assert decode(b'', label) == ('', lookup(label))
         assert encode('', label) == b''
         for repeat in [0, 1, 12]:
             output, _ = iter_decode([b''] * repeat, label)
@@ -74,25 +74,25 @@ def test_invalid_label():
 
 
 def test_decode():
-    assert decode(b'\x80', 'latin1') == '€'
-    assert decode(b'\x80', lookup('latin1')) == '€'
-    assert decode(b'\xc3\xa9', 'utf8') == 'é'
-    assert decode(b'\xc3\xa9', UTF8) == 'é'
-    assert decode(b'\xc3\xa9', 'ascii') == 'Ã©'
-    assert decode(b'\xEF\xBB\xBF\xc3\xa9', 'ascii') == 'é'  # UTF-8 with BOM
+    assert decode(b'\x80', 'latin1') == ('€', lookup('latin1'))
+    assert decode(b'\x80', lookup('latin1')) == ('€', lookup('latin1'))
+    assert decode(b'\xc3\xa9', 'utf8') == ('é', lookup('utf8'))
+    assert decode(b'\xc3\xa9', UTF8) == ('é', lookup('utf8'))
+    assert decode(b'\xc3\xa9', 'ascii') == ('Ã©', lookup('ascii'))
+    assert decode(b'\xEF\xBB\xBF\xc3\xa9', 'ascii') == ('é', lookup('utf8'))  # UTF-8 with BOM
 
-    assert decode(b'\xFE\xFF\x00\xe9', 'ascii') == 'é'  # UTF-16-BE with BOM
-    assert decode(b'\xFF\xFE\xe9\x00', 'ascii') == 'é'  # UTF-16-LE with BOM
-    assert decode(b'\xFE\xFF\xe9\x00', 'ascii') == '\ue900'
-    assert decode(b'\xFF\xFE\x00\xe9', 'ascii') == '\ue900'
+    assert decode(b'\xFE\xFF\x00\xe9', 'ascii') == ('é', lookup('utf-16be'))  # UTF-16-BE with BOM
+    assert decode(b'\xFF\xFE\xe9\x00', 'ascii') == ('é', lookup('utf-16le'))  # UTF-16-LE with BOM
+    assert decode(b'\xFE\xFF\xe9\x00', 'ascii') == ('\ue900', lookup('utf-16be'))
+    assert decode(b'\xFF\xFE\x00\xe9', 'ascii') == ('\ue900', lookup('utf-16le'))
 
-    assert decode(b'\x00\xe9', 'UTF-16BE') == 'é'
-    assert decode(b'\xe9\x00', 'UTF-16LE') == 'é'
-    assert decode(b'\xe9\x00', 'UTF-16') == 'é'
+    assert decode(b'\x00\xe9', 'UTF-16BE') == ('é', lookup('utf-16be'))
+    assert decode(b'\xe9\x00', 'UTF-16LE') == ('é', lookup('utf-16le'))
+    assert decode(b'\xe9\x00', 'UTF-16') == ('é', lookup('utf-16le'))
 
-    assert decode(b'\xe9\x00', 'UTF-16BE') == '\ue900'
-    assert decode(b'\x00\xe9', 'UTF-16LE') == '\ue900'
-    assert decode(b'\x00\xe9', 'UTF-16') == '\ue900'
+    assert decode(b'\xe9\x00', 'UTF-16BE') == ('\ue900', lookup('utf-16be'))
+    assert decode(b'\x00\xe9', 'UTF-16LE') == ('\ue900', lookup('utf-16le'))
+    assert decode(b'\x00\xe9', 'UTF-16') == ('\ue900', lookup('utf-16le'))
 
 
 def test_encode():
@@ -149,5 +149,5 @@ def test_x_user_defined():
     decoded = '2,\x0c\x0b\x1aO\uf7d9#\uf7cb\x0f\uf7c9\uf7bbt\uf7cf\uf7a8\uf7ca'
     encoded = b'aa'
     decoded = 'aa'
-    assert decode(encoded, 'x-user-defined') == decoded
+    assert decode(encoded, 'x-user-defined') == (decoded, lookup('x-user-defined'))
     assert encode(decoded, 'x-user-defined') == encoded

